tests/generator/test_common.py

```diff
@@ -685,6 +685,48 @@ def test_nameserver(self):
 Domains=lab kitchen
 '''})
 
+    def test_mdns(self):
+        self.generate('''network:
+  version: 2
+  ethernets:
+    eth0:
+      mdns: resolve
+    eth1:
+      mdns: true
+    eth2:
+      mdns: false
+''')
+
+        self.assert_networkd({'eth0.network': '''[Match]
+Name=eth0
+
+[Network]
+LinkLocalAddressing=ipv6
+MulticastDNS=resolve
+''',
+                              'eth1.network': '''[Match]
+Name=eth1
+
+[Network]
+LinkLocalAddressing=ipv6
+MulticastDNS=yes
+''',
+                              'eth2.network': '''[Match]
+Name=eth2
+
+[Network]
+LinkLocalAddressing=ipv6
+MulticastDNS=no
+'''})
```

Comment:
```suggestion
        self.assert_networkd({'eth0.network': (ND_EMPTY % ('eth0', 'ipv6'))
                              .replace('ConfigureWithoutCarrier=yes', 'MulticastDNS=resolve'),
                              'eth1.network': (ND_EMPTY % ('eth1', 'ipv6'))
                              .replace('ConfigureWithoutCarrier=yes', 'MulticastDNS=yes'),
                              'eth2.network': (ND_EMPTY % ('eth2', 'ipv6'))
                              .replace('ConfigureWithoutCarrier=yes', 'MulticastDNS=no')})
```
Trying to shrink this a bit...